src/wp-includes/wp-db.php

```diff
@@ -1327,11 +1327,12 @@ public function escape( $data ) {
 	 * @param string $string String to escape.
 	 */
 	public function escape_by_ref( &$string ) {
-		if ( ! is_float( $string ) ) {
+		if ( ! is_float( $string ) && ! is_a( $string, WP_DB_Partial_Query::class ) ) {
```

Comment:
```suggestion
		if ( ! is_float( $string ) && ! $string instanceof WP_DB_Partial_Query ) {
```

`instanceof` is a more performant option than `is_a()`.
When dealing with the database, any performance improvements are worth making.

See https://3v4l.org/93N1d